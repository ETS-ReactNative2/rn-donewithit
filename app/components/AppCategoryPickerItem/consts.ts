const CONSTS = {
  ICON_NAME: 'email',
  TEXT_INPUT_FONT_SIZE: 18,
  ICON_SIZE: 38,
  ICON_WRAP_SIZE: 80
}

export default CONSTS
