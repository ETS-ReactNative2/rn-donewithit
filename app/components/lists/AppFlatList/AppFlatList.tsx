import React from 'react'
import { FlatList, View } from 'react-native'
import styles from './styles'
import CONSTS from './consts'
import { AppFlatListProps } from './types'

function FlatListItemSeparator(hasItemSeparator: boolean) {
  return hasItemSeparator ? <View style={styles.separator} /> : null
}

function AppFlatList({
  ItemSeparatorComponent,
  hasItemSeparator = CONSTS.HAS_ITEM_SEPARATOR,
  ...otherProps
}: AppFlatListProps) {
  return (
    <FlatList
      ItemSeparatorComponent={() => {
        return FlatListItemSeparator(hasItemSeparator)
      }}
      {...otherProps}
    />
  )
}

export default AppFlatList
