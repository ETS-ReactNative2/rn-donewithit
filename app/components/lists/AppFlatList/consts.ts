const CONSTS = {
  HAS_ITEM_SEPARATOR: true
}

export default CONSTS
