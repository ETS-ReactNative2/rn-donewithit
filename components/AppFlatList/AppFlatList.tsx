import React from 'react'
import { FlatList, View, StyleSheet, FlatListProps } from 'react-native'
import COLORS from '../../config/colors'

const styles = StyleSheet.create({
  separator: {
    width: '100%',
    height: 1,
    backgroundColor: COLORS.LIGHT_GRAY
  }
})

function AppFlatList(props: FlatListProps<any>) {
  return (
    <FlatList
      {...props}
      ItemSeparatorComponent={() => <View style={styles.separator} />}
    ></FlatList>
  )
}

export default AppFlatList
