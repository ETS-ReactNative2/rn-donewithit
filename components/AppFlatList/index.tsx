import AppFlatList from './AppFlatList'

export default AppFlatList
