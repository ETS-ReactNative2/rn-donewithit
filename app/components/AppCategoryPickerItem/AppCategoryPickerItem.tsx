import React, { ReactDOM, ReactNode, useCallback } from 'react'
import {
  GestureResponderEvent,
  TouchableOpacity,
  View,
  StyleSheet
} from 'react-native'
import { AppPickerItemProps } from '../../components/AppPickerItem'
import AppText from '../../components/AppText'
import AppIcon from '../AppIcon'
import styles from './styles'
import COLORS from '../../config/colors'
import CONSTS from './consts'
function AppCategoryPickerItem({
  label,
  selected,
  onPress,
  data
}: AppPickerItemProps): JSX.Element {
  const {
    iconName = CONSTS.ICON_NAME,
    iconColor = COLORS.WHITE,
    iconSize = CONSTS.ICON_SIZE,
    iconBackgroundColor = COLORS.PRIMARY
  } = data
  /**
    "iconBackgroundColor": "lightslategrey",
    "iconName": "application-outline",
    "label": "Other",
    "value": 9,
   */
  const customStyles = StyleSheet.create({
    iconWrap: {
      backgroundColor: iconBackgroundColor
    }
  })
  const handlePress = useCallback((e?: GestureResponderEvent) => {
    typeof onPress === 'function' && onPress(e)
  }, [])
  return (
    <TouchableOpacity onPress={handlePress} style={styles.container}>
      <View style={[styles.iconWrap, customStyles.iconWrap]}>
        <AppIcon name={iconName} size={iconSize} color={iconColor} />
      </View>
      <AppText style={[styles.text, selected && styles.selected]}>
        {label}
      </AppText>
    </TouchableOpacity>
  )
}

export default AppCategoryPickerItem
