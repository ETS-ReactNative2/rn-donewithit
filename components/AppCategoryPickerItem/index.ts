import AppCategoryPickerItem from './AppCategoryPickerItem'

export default AppCategoryPickerItem
