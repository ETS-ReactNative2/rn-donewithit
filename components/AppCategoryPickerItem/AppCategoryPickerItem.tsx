import React, { ReactNode, useCallback } from 'react'
import { GestureResponderEvent, TouchableOpacity } from 'react-native'
import { AppPickerItemProps } from '../../components/AppPickerItem'
import AppText from '../../components/AppText'
import styles from './styles'
function AppCategoryPickerItem({
  label,
  selected,
  onPress
}: AppPickerItemProps): ReactNode {
  const handlePress = useCallback((e?: GestureResponderEvent) => {
    typeof onPress === 'function' && onPress(e)
  }, [])
  return (
    <TouchableOpacity onPress={handlePress}>
      <AppText style={[styles.text, selected && styles.selected]}>
        {label}
      </AppText>
    </TouchableOpacity>
  )
}

export default AppCategoryPickerItem
