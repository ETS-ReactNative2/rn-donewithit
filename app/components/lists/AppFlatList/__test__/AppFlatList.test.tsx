import React from 'react'
import renderer from 'react-test-renderer'
import AppFlatList from '../index'
import AppCategoryPickerItem from '../../../../components/AppCategoryPickerItem'
import COLORS from '../../../../config/colors'

jest.mock(
  '@expo/vector-icons/build/vendor/react-native-vector-icons/lib/create-icon-set.js',
  () => {
    return () => ''
  }
)

const props = {
  icon: false,
  placeholder: 'Category',
  options: [
    {
      label: 'Furniture',
      iconName: 'lamp',
      iconBackgroundColor: COLORS.PRIMARY,
      value: 1
    },
    {
      label: 'Cars',
      iconName: 'car',
      iconBackgroundColor: 'orange',
      value: 2
    },
    {
      label: 'Cameras',
      iconName: 'camera',
      iconBackgroundColor: 'gold',
      value: 3
    },
    {
      label: 'Games',
      iconName: 'cards-playing-outline',
      iconBackgroundColor: 'limegreen',
      value: 4
    },
    {
      label: 'Clothing',
      iconName: 'shoe-heel',
      iconBackgroundColor: 'mediumturquoise',
      value: 5
    },
    {
      label: 'Sports',
      iconName: 'basketball',
      iconBackgroundColor: 'cornflowerblue',
      value: 6
    },
    {
      label: 'Movies & Music',
      iconName: 'headphones',
      iconBackgroundColor: 'royalblue',
      value: 7
    },
    {
      label: 'Books',
      iconName: 'book-open-variant',
      iconBackgroundColor: 'mediumorchid',
      value: 8
    },
    {
      label: 'Other',
      iconName: 'application',
      iconBackgroundColor: 'lightslategrey',
      value: 9
    }
  ],
  numColumns: 3,
  hasItemSeparator: false
}

const value = 2

describe('AppFlatList', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <AppFlatList
          numColumns={props.numColumns}
          hasItemSeparator={props.hasItemSeparator}
          data={props.options}
          keyExtractor={(item) => item.value.toString()}
          renderItem={({ item }) => {
            return (
              <AppCategoryPickerItem
                selected={value === item.value}
                label={item.label}
                onPress={() => {
                  console.log(item)
                }}
                data={item}
              />
            )
          }}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
