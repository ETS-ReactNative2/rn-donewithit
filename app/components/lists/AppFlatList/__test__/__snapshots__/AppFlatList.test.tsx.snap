// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppFlatList renders correctly 1`] = `
<RCTScrollView
  ItemSeparatorComponent={[Function]}
  data={
    Array [
      Object {
        "iconBackgroundColor": "tomato",
        "iconName": "lamp",
        "label": "Furniture",
        "value": 1,
      },
      Object {
        "iconBackgroundColor": "orange",
        "iconName": "car",
        "label": "Cars",
        "value": 2,
      },
      Object {
        "iconBackgroundColor": "gold",
        "iconName": "camera",
        "label": "Cameras",
        "value": 3,
      },
      Object {
        "iconBackgroundColor": "limegreen",
        "iconName": "cards-playing-outline",
        "label": "Games",
        "value": 4,
      },
      Object {
        "iconBackgroundColor": "mediumturquoise",
        "iconName": "shoe-heel",
        "label": "Clothing",
        "value": 5,
      },
      Object {
        "iconBackgroundColor": "cornflowerblue",
        "iconName": "basketball",
        "label": "Sports",
        "value": 6,
      },
      Object {
        "iconBackgroundColor": "royalblue",
        "iconName": "headphones",
        "label": "Movies & Music",
        "value": 7,
      },
      Object {
        "iconBackgroundColor": "mediumorchid",
        "iconName": "book-open-variant",
        "label": "Books",
        "value": 8,
      },
      Object {
        "iconBackgroundColor": "lightslategrey",
        "iconName": "application",
        "label": "Other",
        "value": 9,
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "flex-start",
              "marginVertical": 10,
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "tomato",
                  "borderRadius": 40,
                  "height": 80,
                  "justifyContent": "center",
                  "width": 80,
                },
                Object {
                  "backgroundColor": "tomato",
                },
              ]
            }
          >
            <Text />
          </View>
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontFamily": "Courier",
                  "fontSize": 13,
                },
                Array [
                  Object {
                    "fontSize": 18,
                    "marginTop": 6,
                    "textAlign": "center",
                    "width": 80,
                  },
                  false,
                ],
              ]
            }
          >
            Furniture
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "flex-start",
              "marginVertical": 10,
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "tomato",
                  "borderRadius": 40,
                  "height": 80,
                  "justifyContent": "center",
                  "width": 80,
                },
                Object {
                  "backgroundColor": "orange",
                },
              ]
            }
          >
            <Text />
          </View>
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontFamily": "Courier",
                  "fontSize": 13,
                },
                Array [
                  Object {
                    "fontSize": 18,
                    "marginTop": 6,
                    "textAlign": "center",
                    "width": 80,
                  },
                  Object {
                    "color": "tomato",
                  },
                ],
              ]
            }
          >
            Cars
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "flex-start",
              "marginVertical": 10,
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "tomato",
                  "borderRadius": 40,
                  "height": 80,
                  "justifyContent": "center",
                  "width": 80,
                },
                Object {
                  "backgroundColor": "gold",
                },
              ]
            }
          >
            <Text />
          </View>
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontFamily": "Courier",
                  "fontSize": 13,
                },
                Array [
                  Object {
                    "fontSize": 18,
                    "marginTop": 6,
                    "textAlign": "center",
                    "width": 80,
                  },
                  false,
                ],
              ]
            }
          >
            Cameras
          </Text>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "flex-start",
              "marginVertical": 10,
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "tomato",
                  "borderRadius": 40,
                  "height": 80,
                  "justifyContent": "center",
                  "width": 80,
                },
                Object {
                  "backgroundColor": "limegreen",
                },
              ]
            }
          >
            <Text />
          </View>
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontFamily": "Courier",
                  "fontSize": 13,
                },
                Array [
                  Object {
                    "fontSize": 18,
                    "marginTop": 6,
                    "textAlign": "center",
                    "width": 80,
                  },
                  false,
                ],
              ]
            }
          >
            Games
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "flex-start",
              "marginVertical": 10,
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "tomato",
                  "borderRadius": 40,
                  "height": 80,
                  "justifyContent": "center",
                  "width": 80,
                },
                Object {
                  "backgroundColor": "mediumturquoise",
                },
              ]
            }
          >
            <Text />
          </View>
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontFamily": "Courier",
                  "fontSize": 13,
                },
                Array [
                  Object {
                    "fontSize": 18,
                    "marginTop": 6,
                    "textAlign": "center",
                    "width": 80,
                  },
                  false,
                ],
              ]
            }
          >
            Clothing
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "flex-start",
              "marginVertical": 10,
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "tomato",
                  "borderRadius": 40,
                  "height": 80,
                  "justifyContent": "center",
                  "width": 80,
                },
                Object {
                  "backgroundColor": "cornflowerblue",
                },
              ]
            }
          >
            <Text />
          </View>
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontFamily": "Courier",
                  "fontSize": 13,
                },
                Array [
                  Object {
                    "fontSize": 18,
                    "marginTop": 6,
                    "textAlign": "center",
                    "width": 80,
                  },
                  false,
                ],
              ]
            }
          >
            Sports
          </Text>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "flex-start",
              "marginVertical": 10,
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "tomato",
                  "borderRadius": 40,
                  "height": 80,
                  "justifyContent": "center",
                  "width": 80,
                },
                Object {
                  "backgroundColor": "royalblue",
                },
              ]
            }
          >
            <Text />
          </View>
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontFamily": "Courier",
                  "fontSize": 13,
                },
                Array [
                  Object {
                    "fontSize": 18,
                    "marginTop": 6,
                    "textAlign": "center",
                    "width": 80,
                  },
                  false,
                ],
              ]
            }
          >
            Movies & Music
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "flex-start",
              "marginVertical": 10,
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "tomato",
                  "borderRadius": 40,
                  "height": 80,
                  "justifyContent": "center",
                  "width": 80,
                },
                Object {
                  "backgroundColor": "mediumorchid",
                },
              ]
            }
          >
            <Text />
          </View>
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontFamily": "Courier",
                  "fontSize": 13,
                },
                Array [
                  Object {
                    "fontSize": 18,
                    "marginTop": 6,
                    "textAlign": "center",
                    "width": 80,
                  },
                  false,
                ],
              ]
            }
          >
            Books
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "flex-start",
              "marginVertical": 10,
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "tomato",
                  "borderRadius": 40,
                  "height": 80,
                  "justifyContent": "center",
                  "width": 80,
                },
                Object {
                  "backgroundColor": "lightslategrey",
                },
              ]
            }
          >
            <Text />
          </View>
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontFamily": "Courier",
                  "fontSize": 13,
                },
                Array [
                  Object {
                    "fontSize": 18,
                    "marginTop": 6,
                    "textAlign": "center",
                    "width": 80,
                  },
                  false,
                ],
              ]
            }
          >
            Other
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
